woocommerce/wc-admin
# Consume Remote Free Extensions from UI

Fixes https://github.com/woocommerce/woocommerce-admin/issues/6926

Gather a list of free extensions from `/wc-admin/onboarding/free-extensions` which is derived from a woocommerce.com driven config when a merchant is opted into Marketplace Suggestions.

### Detailed test instructions:

First, ensure the default experience is unchanged.

1. Set option `woocommerce_show_marketplace_suggestions` to "no" to disable the remote config.
2. Go to Business Details of the OBW, `/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=business-features`.
3. Click the "Free features" tab.
4. This should be unchanged from the current implementation. Poke around and make sure nothing is broken.

Next, test the remote endpoint

1. Turn on Marketplace Suggestions by setting `woocommerce_show_marketplace_suggestions` to "yes"
2. Install the Remote Tester to add in the data feed because it doesn't exist yet on wccom. Use [the branch found in this pull request](https://github.com/joshuatf/woocommerce-admin-remote-tester/pull/3) if it hasn't yet been merged.
3. Go to Business Details of the OBW, `/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=business-features`.
4. Click the "Free features" tab.
5. See the list populated by [the config](https://github.com/joshuatf/woocommerce-admin-remote-tester/blob/0f9aeba29c38e153f803aa4464e907c679480fa7/free-extensions.json) which only has two extensions. Poke around and make sure the functionality is the same.
6. If you don't see either of the plugin, make sure they aren't activated. The new config rules only show plugins that aren't already installed/activated.

<img width="625" alt="Screen Shot 2021-05-07 at 2 26 21 PM" src="https://user-images.githubusercontent.com/1922453/117389448-7fb0e800-af40-11eb-9654-ee237dfad5a8.png">
